src/test/java/com/limechain/sync/fullsync/BlockExecutorIsolatedTest.java

```diff
@@ -0,0 +1,119 @@
+package com.limechain.sync.fullsync;
+
+import com.google.protobuf.ByteString;
+import com.limechain.chain.spec.ChainSpec;
+import com.limechain.network.protocol.blockannounce.scale.BlockHeaderScaleWriter;
+import com.limechain.network.protocol.warp.dto.Block;
+import com.limechain.network.protocol.warp.dto.BlockBody;
+import com.limechain.network.protocol.warp.dto.BlockHeader;
+import com.limechain.network.protocol.warp.dto.Extrinsics;
+import com.limechain.network.protocol.warp.dto.HeaderDigest;
+import com.limechain.network.protocol.warp.scale.reader.HeaderDigestReader;
+import com.limechain.network.protocol.warp.scale.writer.BlockBodyWriter;
+import com.limechain.runtime.research.hybrid.WasmRuntimeInstance;
+import com.limechain.runtime.research.hybrid.WasmRuntimeInstanceBuilder;
+import com.limechain.runtime.research.hybrid.context.Context;
+import com.limechain.runtime.research.hybrid.trieaccessor.TrieAccessor;
+import com.limechain.runtime.version.StateVersion;
+import com.limechain.storage.KVRepository;
+import com.limechain.storage.trie.TrieStorage;
+import com.limechain.trie.TrieStructureFactory;
+import com.limechain.utils.StringUtils;
+import com.limechain.utils.scale.ScaleUtils;
+import com.limechain.utils.scale.readers.PairReader;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import io.emeraldpay.polkaj.scale.reader.ListReader;
+import io.emeraldpay.polkaj.types.Hash256;
+import lombok.extern.java.Log;
+import org.apache.commons.lang3.ArrayUtils;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Log
+class BlockExecutorIsolatedTest {
```

Comment:
With this test not including booting the Spring app we lose the Spring logger, so logs are a bit uglier (and all in red for some reason). Think about whether this is an issue we'd want to investigate.